primitives/src/identity_or_key.rs

```diff
@@ -0,0 +1,85 @@
+use crate::identity_id::IdentityId;
```

Comment:
@maxsam4 This concept is similar to `Signer` struct ( in `primitives/src/signing_item.rs`). 
Could we integrate both into just one?

Reply:
Yeah, that makes sense. I'll merge the two.